sql/analyzer/apply_hash_in.go

```diff
@@ -57,7 +57,7 @@ func hasSingleOutput(e sql.Expression) bool {
 		switch expr.(type) {
 		case expression.Tuple, *expression.Literal, *expression.GetField,
 			expression.Comparer, *expression.Convert, sql.FunctionExpression,
-			*expression.IsTrue, *expression.IsNull, *expression.Arithmetic:
+			*expression.IsTrue, *expression.IsNull, expression.ArithmeticOp:
```

Comment:
It might be worth adding some unique method to the new `ArithmeticOp` interface, so that we can be confident only the Arithmetic operations will conform to it. I think it's worth seeing if we can simplify to use `BinaryExpression` instead of `sql.Expression` in the interface definition, but that would also mean that any binary expression would conform to the new interface, and that probably isn't what we want here. I'd recommend adding an arithmetic-specific function like `Operator() string` to the interface; that could be an easy way to make sure only the arithmetic types conform to that interface, plus it seems like it makes sense for the interface to provide something like that.